src/resolve.rs

```diff
@@ -395,55 +416,198 @@ mod toml {
 #[cfg(test)]
 mod tests {
     use super::{Resolve, ResolveError, ResolveMut};
-    use crate::Pointer;
+    use crate::{
+        index::{OutOfBoundsError, ParseIndexError},
+        Pointer,
+    };
     use core::fmt;
 
-    struct Test<'v, V> {
-        ptr: &'static str,
-        expected_result: Result<&'v V, ResolveError>,
-        data: &'v V,
+    #[cfg(feature = "std")]
+    #[test]
+    fn resolve_error_source() {
```

Comment:
Not sure I get the point of this test. Not a problem per se, but feels like it'll just cause maintenance burden down the line.